x-pack/functionbeat/tests/system/test_cloudwatch.py

```diff
@@ -0,0 +1,284 @@
+        s3 = boto3.resource('s3')
+        bucket = s3.Bucket(self.bucket)
+        for key in bucket.objects.all():
+            key.delete()
+        bucket.delete()
+
+    @unittest.skipIf(not FUNCTIONBEAT_INTEGRATION_TESTS,
+                     "functionbeat integration tests are disabled, run with FUNCTIONBEAT_INTEGRATION_TESTS=1 to enable them.")
+    def test_deploy_cloudwatch_local_options(self):
+        """
+        Deploy a function to retrieve cloudwatch logs.
+        """
+
+        # Setup the function
+        self.deploy(
+            deploy_bucket=self.bucket,
+            function={
+                'name': self.function_name,
+                'enabled': True,
+                'type': 'cloudwatch_logs',
+                'triggers': [
+                    {'log_group_name': self.log_group_name},
+                ],
+                'processors': [
+                    {'dissect': {'tokenizer': 'id=%{id} m=%{message}'}},
+                ],
+                'fields': {
+                    'hello': 'world',
+                }
+            },
+            cloud_id=self.cloud_id(),
+            cloud_auth=self.cloud_auth(),
+        )
+
+        uid = uuid.uuid4()
+        # Insert some events into cloudwatch logs
+        self.client.put_log_events(logGroupName=self.log_group_name,
+                                   logStreamName=self.log_stream_name,
+                                   logEvents=[
+                                       {
+                                           'timestamp': time_now_millis(),
+                                           'message': 'id=%s m=hello world 1' % uid
+                                       },
+                                       {
+                                           'timestamp': time_now_millis(),
+                                           'message': 'id=%s m=hello world 2' % uid
+                                       },
+                                       {
+                                           'timestamp': time_now_millis(),
+                                           'message': 'id=%s m=hello world 3' % uid
+                                       },
+                                       {
+                                           'timestamp': time_now_millis(),
+                                           'message': 'id=%s m=hello world 4' % uid
+                                       },
+                                       {
+                                           'timestamp': time_now_millis(),
+                                           'message': 'id=%s m=hello world 5' % uid
+                                       },
+                                   ])
+
+        self.wait_until(lambda: self.es.search(index="functionbeat*", body={"query": {"match_all": {}}})['hits']['total'] >= 5,
+                        max_timeout=TIMEOUT)
+        results = self.es.search(index="functionbeat*")
```

Comment:
If possible have the test case + some random part in the index name. In case we ever run tests in parallel they might interfere otherwise (e.g. if test cases for different branches are executed at the same time).